packages/gestalt/src/SlimBanner.js

```diff
@@ -202,15 +202,18 @@ export default function SlimBanner({
 
         <Flex.Item flex="grow">
           <Box dangerouslySetInlineStyle={{ __style: !isDefault ? { marginTop: '-1px' } : {} }}>
-            <Text inline>
-              {message}
-              {helperLink ? (
-                <Fragment>
-                  {' '}
-                  <HelperLink {...helperLink} />
-                </Fragment>
-              ) : null}
-            </Text>
+            {typeof message === 'string' ? (
+              <Text inline>
+                {message}
+                {helperLink ? (
+                  <Fragment>
+                    {' '}
+                    <HelperLink {...helperLink} />
+                  </Fragment>
+                ) : null}
+              </Text>
+            ) : null}
+            {typeof message !== 'string' && isValidElement(message) ? message : null}
```

Comment:
Do we want to enforce the usage of Text here? Otherwise the user can pass any valid React element here and suppress the Flow error. I think we might be able to use `message.type.name === 'Text'` — the user would be able to create their own Text component and pass that, but that seems a lot less likely (and much more intentional).

Reply:
Im going to try using React.children to make that check. I think it can be used with any prop

Reply:
Ah ok — I haven't actually tried this technique in production code, so I wasn't sure if it would work. Make the best attempt you can…while throwing an error here on bad input would be ideal, it's not a dealbreaker if we can't and this current technique is the best we can do.

Reply:
Misplaced the reply https://github.com/pinterest/gestalt/pull/2484#discussion_r1019395689

Reply:
Fixed it by adding .displayName to 'Text'

Reply:
Nice! 🔥